include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+
+/**
+ *
+ * return the name of a kind
+ *
+ * kind cannot be null
+ */
+const char *rp_kind_get_name(rp_kind *kind);
+
+/**
+ * creates a target from the provided info
+ *
+ * kind cannot be null
+ * path_components_count must be equal to the granularity depth of the kind
+ * path_components size must be equal to path_components_count
+ *
+ * returns 0 if a error was encountered, 1 otherwise.
+ */
+rp_target *rp_create_target(rp_kind *kind,
+                            int is_exact,
+                            uint64_t path_components_count,
+                            const char *path_components[]);
+
+/**
+ * Deletes the provided target, (THIS TARGET MUST BE ONE OBTAINED FROM
+ * rp_create_target)
+ */
+void rp_destroy_target(rp_target *target);
+
+/**
+ * requests the production of the provided targets in a particular container
+ *
+ * returns 0 if an error was encountered, 1 otherwise
+ *
+ * runner cannot be null
+ * target cannot be null
+ * container_name cannot be null
+ */
+bool rp_produce_targets(rp_runner *runner,
+                        uint64_t targets_count,
+                        rp_target *targets[],
+                        rp_step *step,
+                        rp_container_checkpoint *container);
+
+/**
+ * serialize the provided container at the provided step in the provided path
+ *
+ * no argument can be null
+ *
+ * returns 0 if an error was encountered 1 otherwise
+ */
+bool rp_container_checkpoint_store(rp_container_checkpoint *container,
+                                   const char *path);
+
+/**
+ * load the provided container at the provided step from the provided path
+ *
+ * no argument can be null
+ *
+ * return 0 if a error was encountered 1 otherwise
+ *
+ */
+bool rp_container_checkpoint_load(rp_container_checkpoint *container,
+                                  const char *path);
+
+/**
+ * returns a table containing all the scalar available target that can
+ * be requested to the runner
+ *
+ * runner cannot be null
+ */
+void rp_runner_recompute_all_available_targets(rp_runner *runner);
+
+/**
+ * returns the container status associated to the provided checkpoint
+ * or null if no status is associated to the provided checkpoint
+ *
+ * no argument can be null
+ */
+rp_container_status *
+rp_get_container_status(rp_runner *runner, rp_container_checkpoint *container);
+
+/**
+ * returns the amount of target inside the provided container statuses
+ */
+uint64_t rp_container_status_count(rp_container_status *status);
+
+/**
+ *  returns the nth target inside the provided status or null if it's out of
+ * bounds
+ *
+ *  status cannot be null
+ */
+rp_target *
+rp_container_status_get_target(rp_container_status *status, uint64_t index);
+
+/**
+ * returns the kind of the provided target
+ *
+ * target cannot be null
+ */
+rp_kind *rp_target_get_kind(rp_target *target);
+
+/**
+ *
+ * Serializes target into a string, target cannot be null.
+ *
+ */
+char *rp_create_target_serialization(rp_target *target);
```

Comment:
```suggestion
char *rp_create_target_serialize(rp_target *target);
```
or
```suggestion
char *rp_create_target_to_string(rp_target *target);
```
* Who owns the return value? Should it be `const char *`?
* This function starts with `rp_create_` but has no `rp_destroy_` counterpart. Let's be strict on this.